core/xml.js

```diff
@@ -326,11 +326,8 @@ Blockly.Xml.domToPrettyText = function(dom) {
  */
 Blockly.Xml.textToDom = function(text) {
   var doc = Blockly.Xml.utils.textToDomDocument(text);
-  // This function only accepts <xml> documents.
-  if (!doc || !doc.documentElement ||
-      doc.documentElement.nodeName.toLowerCase() != 'xml') {
-    // Whatever we got back from the parser is not the expected structure.
-    throw TypeError('Blockly.Xml.textToDom expected an <xml> document.');
```

Comment:
A couple of issues here.

First, the JSDoc on this function no longer describes its operation ("It requires the XML to have a root element of \<xml\>.").

Second, the return value is no longer guaranteed to be an XML node.  There are 11 callers of this function that might object to getting a non-XML node (I'm seeing a lot of things like `Blockly.Xml.textToDom(text).firstChild` which presumes that a wrapping XML node is guaranteed).

Reply:
Oh woops! Doc is now fixed.

With regard to your second point, I don't think that should break anything. Anything expecting an XML node will get an XML node because `<xml></xml>` tags were passed into the function. The only change is that in the future, calls to domToBlock and similar functions won't have to wrap there text in `<xml></xml>` tags like they did in the past.

However if you have any suggestions for changes I'm happy to impliment.